src/tokenizer/types.js

```diff
@@ -45,13 +58,13 @@ export const types = {
   eof: new TokenType("eof"),
 
   // Punctuation token types.
-  bracketL: new TokenType("[", {beforeExpr: true, startsExpr: true}),
+  bracketL: new TokenType("[", { ...beforeExpr, ...startsExpr }),
```

Comment:
Seems there are a lot of cases where we have `before` and `startsExpr` maybe we can do the spread in this case only once?

```js
const beforeAndStartExpr = { ...beforeExpr, ...startsExpr };
```